src/frontend/dash/index.ts

```diff
@@ -14,31 +14,41 @@ refreshDelphi()
 setInterval(refreshDelphi, 1000 * 60 * 2) // refresh post every 1 minute
 
 function regenCircles(loggedin: LoggedIn) {
-    const desiredRatio = 1; // y / x
-    const ratioError = .1;
-    let tries = 0;
-    let { maxX, maxY, minX, minY } = getBounds()
     let placedCircles: MemberCircle[] = [];
-    do {
-        tries++;
-        const circles = []
-        const now = Date.now()
-        Object.entries(loggedin).forEach(ent => {
-            const member = members.find(o => o.name == ent[0])
-            circles.push(new MemberCircle(
-                (now - ent[1]) / 1000 / 60 / 60,
-                member.firstname,
-                member.img
-            ))
-        });
-        placedCircles = placeCircles(circles);
 
-        ({ maxX, maxY, minX, minY } = getBounds());
+    const circles = []
+    const now = Date.now()
+    for(const ent in loggedin) {
+        const member = members.find(o => o.name == ent)
+        circles.push(new MemberCircle(
+            (now - loggedin[ent]) / 360000, // 1000 / 60 / 60
+            member.firstname,
+            member.img
+        ))
+    }
 
-    } while (tries < 1000 && ((maxY - minY) / (maxX - minX) < (desiredRatio - ratioError / desiredRatio) || (maxY - minY) / (maxX - minX) > (desiredRatio + ratioError / desiredRatio))) // test to make sure the dimentions are chill
+    setDelphiVisibility(getNameDensity(circles) < 1)
+
+    setAspectRatio(getRatio());
+    placedCircles = placeCircles(circles);
     redrawCircles(placedCircles)
 }
 
+const densityMultiplier = 0.006;
```

Comment:
Where'd this constant come from?

Reply:
If you're asking how I came up with the value, it probably needs to be tweak for the dashboard in the lab

Reply:
But how'd you come up with this value?

Reply:
Is there any kind of process to it? Is it somehow related to the screen size or is it completely arbitrary?